crates/predictor/src/predictor/dataset_reader.rs

```diff
@@ -89,11 +90,30 @@ impl UninitializedDataSetReader {
                     }
                 };
 
-                let mut readers : Vec<Box<GribReader>> = vec![];
-                for file in bucket {
+                let mut readers : HashMap<i32, Box<GribReader>> = HashMap::new();
+                let mut last_hour = 0.0;
+                let mut last_file = &bucket[0];
+
+                for file in &bucket[1..] {
                     println!("{}", file.path().display());
 
-                    readers.push(Box::new(GribReader::new(file.path().to_str().unwrap().to_string())));
+                    let name = file.file_name().into_string().unwrap();
+                    let hour = name.split("_").collect::<Vec<&str>>()[4][0..3].parse::<f32>().unwrap();
```

Comment:
A little hard to tell what's going on without knowing the file format. Add a short comment explaining what's going on